pkg/server/server_test.go

```diff
@@ -486,6 +488,127 @@ func TestCheckWithCachedResolution(t *testing.T) {
 	require.True(t, checkResponse.Allowed)
 }
 
+func TestWriteAndReadAssertionModelDSError(t *testing.T) {
+	ctx := context.Background()
+
+	storeID := ulid.Make().String()
+	modelID := ulid.Make().String()
+
+	typedefs := parser.MustParse(`
+	type user
+
+	type repo
+	  relations
+	    define reader: [user] as self
+	`)
+
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+
+	mockDSOldSchema := mockstorage.NewMockOpenFGADatastore(mockController)
+
+	mockDSOldSchema.EXPECT().
+		ReadAuthorizationModel(gomock.Any(), storeID, modelID).
+		AnyTimes().
+		Return(&openfgav1.AuthorizationModel{
+			SchemaVersion:   typesystem.SchemaVersion1_0,
+			TypeDefinitions: typedefs,
+		}, nil)
+
+	mockDSBadReadAuthModel := mockstorage.NewMockOpenFGADatastore(mockController)
+
+	mockDSBadReadAuthModel.EXPECT().
+		ReadAuthorizationModel(gomock.Any(), storeID, modelID).
+		AnyTimes().
+		Return(nil, fmt.Errorf("unable to read"))
+
+	mockDSBadWriteAssertions := mockstorage.NewMockOpenFGADatastore(mockController)
+	mockDSBadWriteAssertions.EXPECT().
+		ReadAuthorizationModel(gomock.Any(), storeID, modelID).
+		AnyTimes().
+		Return(&openfgav1.AuthorizationModel{
+			SchemaVersion:   typesystem.SchemaVersion1_1,
+			TypeDefinitions: typedefs,
+		}, nil)
+	mockDSBadWriteAssertions.EXPECT().
+		WriteAssertions(gomock.Any(), storeID, modelID, gomock.Any()).
+		AnyTimes().
+		Return(fmt.Errorf("unable to write"))
+
+	tests := []struct {
+		name          string
+		assertions    []*openfgav1.Assertion
+		mockDatastore *mockstorage.MockOpenFGADatastore
+		expectedError error
+	}{
+		{
+			name:          "unsupported_schema",
+			assertions:    []*openfgav1.Assertion{},
+			mockDatastore: mockDSOldSchema,
+			expectedError: serverErrors.ValidationError(
+				fmt.Errorf("invalid schema version"),
+			),
+		},
+		{
+			name:          "failed_to_read",
+			assertions:    []*openfgav1.Assertion{},
+			mockDatastore: mockDSBadReadAuthModel,
+			expectedError: serverErrors.NewInternalError(
+				"", fmt.Errorf("unable to read"),
+			),
+		},
+		{
+			name:          "failed_to_write",
+			assertions:    []*openfgav1.Assertion{},
+			mockDatastore: mockDSBadWriteAssertions,
+			expectedError: serverErrors.NewInternalError(
+				"", fmt.Errorf("unable to write"),
+			),
+		},
+	}
+
+	for _, curTest := range tests {
+		curTest := curTest
```

Comment:
You don't need this line here. This is only needed with parallel tests. See https://gist.github.com/posener/92a55c4cd441fc5e5e85f27bca008721